File: Patient/Patient.cpp
#include "Patient.hpp"

Patient :: Patient()
    :id(""), name(""), gender(""), age(0), procedureName(""), consumables(""), reportIds(""), busy(0) {}

Patient::Patient(const string& Name, const string& Gender, int Age, const string& ProcedureName) 
    : id(""),name(Name), gender(Gender), age(Age), procedureName(ProcedureName), consumables(""), reportIds(""),busy(0) {}

Patient& Patient::operator=(const Patient& p) {
    this->id = p.id;
    this->name = p.name;
    this->gender = p.gender;
    this->age = p.age;
    this->procedureName = p.procedureName;
    this->consumables = p.consumables;
    this->reportIds = p.reportIds;
    this->busy = p.busy;
    return *this;
}

string Patient::getId(){
    return id;
}

string Patient::getName() {
    return name;
}

bool Patient::isEmpty() {
    if (name.empty()) {
        return true;
    }
    return false;
}

string Patient::getGender() {
    return gender;
}

int Patient::getAge() {
    return age;
}

string Patient::getProcedureName() {
    return procedureName;
}

string Patient::getConsumables() {
    return consumables;
}

string Patient::getReportIds() {
    return reportIds;
}

int Patient :: getBusyStatus(){
    return busy;
}

void Patient :: setId(const string& newId){
    id = newId;
}

void Patient::setName(const string& newName) {
    name = newName;
}

void Patient::setGender(const string& newGender) {
    gender = newGender;
}

void Patient::setAge(int newAge) {
    age = newAge;
}

void Patient::setProcedureName(const string& newProcedureName) {
    procedureName = newProcedureName;
}

void Patient::addConsumable(const string& newConsumable) {
    if (consumables.empty()) {
        consumables = newConsumable;
    }
    else {
        consumables = consumables + "," + newConsumable;
    }
}

void Patient::addReportId(const string& newReportId) {
    if (reportIds.empty()) {
        reportIds = newReportId;
    }
    else {
        reportIds = reportIds + "," + newReportId;
    }
}

void Patient::setConsumables(const string& newConsumables)
{
    consumables = newConsumables;
}

void Patient::setReportIds(const string& newReportIds)
{
    reportIds = newReportIds;
}

void Patient::setBusyStatus(int newStatus)
{
    busy = newStatus;
}

string Patient::toString() {
    string result = id + ";" + name + ";" + gender + ";" + to_string(age) + ";" + procedureName + ";" + consumables + ";" + reportIds + ";" + to_string(busy);
    return result;
}

File: Devices/PythonClient/Patient.py
class Patient:
    def __init__(self, name="", age=0, gender="", procedureName=""):
        self.id = 0
        self.name = name
        self.age = age
        self.gender = gender
        self.procedureName = procedureName
        self.consumables = ""
        self.reportIds = ""
        self.busy = 0

    def setId(self, id):
        self.id = id

    def setName(self, name):
        self.name = name

    def setAge(self, age):
        self.age = age

    def setGender(self, gender):
        self.gender = gender

    def setProcedureName(self, procedureName):
        self.procedureName = procedureName

    def setConsumables(self, consumables):
        self.consumables = consumables

    def setReportIds(self, reportIds):
        self.reportIds = reportIds

    def addConsumables(self, consumables):
        if self.consumables == "":
            self.consumables = consumables
        else:
            self.consumables = self.consumables + "," + consumables

    def addReportIds(self, reportIds):
        if self.reportIds == "":
            self.reportIds = reportIds
        else:
            self.reportIds = self.reportIds + "," + reportIds

    def setBusyStatus(self, busyStatus):
        self.busy = busyStatus

    def getId(self):
        return self.id

    def getName(self):
        return self.name

    def getAge(self):
        return self.age

    def getGender(self):
        return self.gender

    def getProcedureName(self):
        return self.procedureName

    def getConsumables(self):
        return self.consumables

    def getReportIds(self):
        return self.reportIds

    def getBusyStatus(self):
        return self.busy

    def __eq__(self, other):
        self.id = other.getId()
        self.name = other.getName()
        self.age = other.getAge()
        self.gender = other.getGender()
        self.procedureName = other.getProcedureName()
        self.consumables = other.getConsumables()
        self.reportIds = other.getReportIds()
        self.busy = other.getBusyStatus()

    def toString(self):
        return str(self.id) + ";" + self.name + ";" + self.gender + ";" + str(self.age) + ";" + self.procedureName + ";" + self.consumables + ";" + self.reportIds + ";" + str(self.busy)
